### PR TITLE
update ironic-image required tests to Prow tests

### PR DESCRIPTION
**Pull request by tuminoid**
Update ironic-image required tests to Prow tests. Remove optional tests as they conflict with required tests, and make little sense to be available cross-releases.

### CODE REVIEW

**Comment by tuminoid** [comment]
/hold

Optional tests needed to be removed as they conflict with required tests. IMO they also make little sense anyways. Thoughts?

/cc @mboukhalfa @kashifest @lentzi90

File: prow/manifests/overlays/metal3/config.yaml
Comment (lentzi90) [inline prow/manifests/overlays/metal3/config.yaml:2928]: These are the actual prow job definitions. Without them you won't be able to trigger anything through prow.
The required status checks above are just things tide should make sure are there before merging. I.e. prow knows nothing about them except that they must be there

**Review by lentzi90 — CHANGES_REQUESTED** [review]
The prow jobs have to stay if we want to trigger the tests through prow.

File: prow/manifests/overlays/metal3/config.yaml
Comment (tuminoid) [reply prow/manifests/overlays/metal3/config.yaml:2928]: Oh yes, of course. However they cannot be optional, and need to be branch by branch defined then, as they cannot be listed like this and be required. Prow won't allow that.

**Comment by tuminoid** [comment]
OK, fixed the confusion with the jenkins contexts and prow contexts. How about now?

**Review by lentzi90 — COMMENTED** [review]
/approve

**Review by Rozzii — APPROVED** [review]
/lgtm

**Pull request merged**
